### ISSUE TITLE
Slabs can result in one-way gates

### ISSUE THREAD

**Issue by SmallJoker** [minetest/minetest#1229]
EDIT:
Using 1/4 slabs from moreblocks, you can build one-way gates or traps.

From outside:
![1](http://i.imgur.com/XOnvlnb.png)
Impossible to walk forwards or sidewards:
![2](http://i.imgur.com/UnMdIRo.png)

Bug found in version 0.4.9, commit 8b02a01 - Windows XP

**Comment by nerzhul** [comment]
@SmallJoker refresh please, and close it if it's not possible to reproduce the problem now

**Comment by Megaf** [comment]
Actually It's quite easy to reproduce the problem.
There is quite a lot of traps in my server exploring this bug.
However, we use toasters from homedecor.

To reproduce, make a three nodes tall column and place another node beside the last node(the tallest).
Place a toaster under it, go on the toaster, you are trapped for life now.

**Comment by est31** [comment]
@SmallJoker can you upload the screenshots to github? This way they get preserved if the image hoster deletest the image, like in this case.

**Comment by RobertZenz** [comment]
I'm unable to reproduce this with the latest HEAD, either you can through, or you can't, the side doesn't matter.

Test setup is the following doorway:

![screenshot_20160117_143237](https://cloud.githubusercontent.com/assets/611377/12377710/3a7179f0-bd27-11e5-96de-ddb554b415cf.png)

I can go through the doorway if the placed slap has a height below or equal to 0.2, but can't get through with a height greater or equal to 0.3.

**Comment by paramat** [comment]
Doesn't seem a bug to me, just a result of being at different starting heights depending on direction. Is it really a problem?

**Comment by SmallJoker** [comment]
It is not really a problem - if it's not abused as a trap, like Megaf said some months ago.
Updated the images in the first post with the must recent build I've got.

@est31 does not have to worry. I uploaded the images in my account. I can delete them myself by accident, so I don't have to wait until they "expire".

**Comment by paramat** [comment]
@SmallJoker shall we close this perhaps?

**Comment by SmallJoker** [comment]
Yes, this is a trivial bug. Let's close this until someone else complains about it ;)

**Comment by Megaf** [comment]
Do we still need this issue open?

**Comment by paramat** [comment]
Good question.

**Comment by MarkuBu** [comment]
I noticed another issue that might be related to this one.

If I walk from a slab like snow to a smaller slab I can see that the player is "sinking" into that slab.

Can check that tomorrow again

**Comment by paramat** [comment]
The slow sink happens in several situations whenever the player is just a little way above a solid node #664

**Comment by paramat** [comment]
https://github.com/minetest/minetest/issues/664#issuecomment-158778984 explains how it happens.

**Comment by paramat** [comment]
> The MTG player is now 1.7 nodes high, so quarter slabs will no longer trap in MTG

Closing.

**Comment by numberZero** [comment]
> The MTG player is now 1.7 nodes high, so quarter slabs will no longer trap in MTG

But this is MT and not MTG issue. I remember I’ve been using `moreblocks` slabs to build one-way gates (not traps!); that should still be possible. (But these were passable in the opposite directions using sneaking). Really, I don’t know if that is a bug or a feature; there are many other one-way constructions possible (involving 3-node fall, etc.).

**Comment by MarkuBu** [comment]
It is definitely a bug. Just because it can be used for something else doen't make it a feature. To make such a feature you may have to make a special "one way node"

Same with "Light through stairs" bug. Just because some people use it for buildings doesn't make it a feature because a lot of people don't want this effect. If you want this effect add a stair node for this

**Comment by SmallJoker** [comment]
This is just moving the issue to another height for slabs to get the same effect.
It seems to be a edge case in the collision calculation, at least for the automatic step-up function.

**Comment by paramat** [comment]
SmallJoker, agreed, sorry.

**Comment by SmallJoker** [comment]
@paramat In this case collision does not work properly. The issue is still existent.

**Comment by numberZero** [comment]
@SmallJoker Can’t reproduce on 5.0.

**Comment by paramat** [comment]
I think collision does work properly and this is just a somewhat inevitable consequence of how it works. 'Fixing' it will likely be very difficult, possibly complex, and seems not worth the effort as it's a minor issue.
I also don't think such a trap is something that necessarily needs to be removed, it can be considered a feature, several people have expressed this above.

Note that no-one finds it an actual practical problem yet, it's just something that people are seeing as a possible problem, as you wrote above (and closed the issue):
> Yes, this is a trivial bug. Let's close this until someone else complains about it ;)

Let's wait until someone proves this is an actual serious problem?
